public/app/features/variables/state/actions.ts

```diff
@@ -222,13 +248,14 @@ export const processVariable = (
         refreshableVariable.refresh === VariableRefresh.onDashboardLoad ||
         refreshableVariable.refresh === VariableRefresh.onTimeRangeChanged
       ) {
-        await variableAdapters.get(variable.type).updateOptions(refreshableVariable);
-        dispatch(resolveInitLock(toVariablePayload(variable)));
+        await dispatch(updateOptions(toVariableIdentifier(refreshableVariable)));
         return;
       }
     }
 
-    dispatch(resolveInitLock(toVariablePayload(variable)));
+    // for variables that aren't updated via url or refresh let's simulate the same state changes
+    dispatch(variableStateFetching(toVariablePayload(variable)));
+    dispatch(variableStateCompleted(toVariablePayload(variable)));
```

Comment:
Why do we need to go via fetching here can't we just emit completed action?

Reply:
It might just be a copy-paste error, will check. Thanks for finding this